File: src/main/java/com/chrisss404/example/Utility.java
/*
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301  USA
 */

package com.chrisss404.example;

import org.w3c.dom.Document;
import org.xml.sax.SAXException;

import javax.xml.parsers.DocumentBuilderFactory;
import javax.xml.parsers.ParserConfigurationException;
import java.io.IOException;
import java.io.InputStream;

/**
 * Contains utility functions.
 */
public final class Utility {

    private Utility() {
    }

    /**
     * Creates a DOM document from a given XML input.
     *
     * @param xmlInputStream The XML input source.
     * @return The generated dom document.
     */
    public static Document loadXmlDocument(InputStream xmlInputStream) {
        try {
            DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
            dbf.setNamespaceAware(true);
            return dbf.newDocumentBuilder().parse(xmlInputStream);
        } catch (ParserConfigurationException | IOException | SAXException e) {
            throw new InvalidDocumentException("Invalid XML input", e);
        }
    }

}

File: src/main/php/Utility.php
<?php
/*
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301  USA
 */

namespace Example;

use TheSeer\fDOM\fDOMDocument;
use TheSeer\fDOM\fDOMException;

final class Utility
{
    /**
     * @param $xmlString string
     * @return fDOMDocument
     */
    public static function loadXmlDocument($xmlString)
    {
        try {
            if (empty($xmlString)) {
                throw new \InvalidArgumentException("Input cannot be empty");
            }

            $domDocument = new fDOMDocument();
            $domDocument->loadXML($xmlString);

            return $domDocument;
        } catch (fDOMException $exception) {
            throw new InvalidDocumentException("Invalid XML input", $exception->getCode(), $exception);
        } catch (\InvalidArgumentException $exception) {
            throw new InvalidDocumentException("Invalid XML input", $exception->getCode(), $exception);
        }
    }
}
